stan/math/prim/functor/coupled_ode_observer.hpp

```diff
@@ -84,7 +85,14 @@ struct coupled_ode_observer {
         N_(y0.size()),
         M_(theta.size()),
         index_offset_theta_(is_constant_all<T1>::value ? 0 : N_ * N_),
-        next_ts_index_(0) {}
+        next_ts_index_(0) {
+    if (!is_constant_all<T_t0>::value) {
+      f_y0_t0_
+          = f(value_of(t0), value_of(y0), value_of(theta_), x_, x_int_, msgs_);
+      check_size_match("coupled_ode_observer", "dy_dt", f_y0_t0_.size(),
```

Comment:
Good catch - we need the check here. Is this check triggered somewhere in the tests?

Reply:
No, but I can add that.

Reply:
Checks added!